mindmeld/active_learning/alp.py

```diff
@@ -104,6 +114,22 @@ def __init__(  # pylint: disable=R0913
         self.init_sampled_queries_ids = None
         self.data_bucket = None
 
+        if not (self.classifier_tuning_strategies or self.tagger_tuning_strategies):
+            raise ValueError("No tuning strategy provided.")
+
+        for level in self.tuning_level:
+            if level not in [TUNE_LEVEL_DOMAIN, TUNE_LEVEL_INTENT, TUNE_LEVEL_ENTITY]:
+                raise ValueError(f"Invalid tuning level: {level}")
+
+        if (
+            TUNE_LEVEL_DOMAIN in self.tuning_level
+            and TUNE_LEVEL_INTENT in self.tuning_level
+        ):
+            logger.info(
```

Comment:
Nit: Perhaps we should raise an error here instead of silently selecting intent on behalf of the user.  A log may go undetected in the terminal. Since AL can take a long time to train, it would be ideal if the user is clear on what training level is selected, especially if they were aiming to train at the domain-level and not at the intent-level.

Reply:
Instead of raising an error, how about creating a warning? While it makes sense that users notice this automatic preference, we should also consider minimal breaking points